### ISSUE TITLE
Integrate location

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/debug#7]
The `location` function in TemplateHaskell can given me the source file, and location of the Debug block. Some kind of smart lexer and matching algorithm could give the original code, precisely, with exact locations and layout as per the original.

**Comment by pepeiborra** [comment]
It would probably be easier to just open the source file and copy the original code verbatim.

**Comment by ndmitchell** [comment]
Yep, that's what I'm suggesting - the remarks about a smart lexer are more about trying to match back assuming variables were not single atomic keywords (which was the original plan, but hasn't really happened, so can be ignored)
